homeassistant/components/group/registry.py

```diff
@@ -8,14 +8,130 @@
 from dataclasses import dataclass
 from typing import Protocol
 
-from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.components.climate import HVACMode
+from homeassistant.components.vacuum import STATE_CLEANING, STATE_ERROR, STATE_RETURNING
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_ELECTRIC,
+    STATE_GAS,
+    STATE_HEAT_PUMP,
+    STATE_HIGH_DEMAND,
+    STATE_PERFORMANCE,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_ARMED_VACATION,
+    STATE_ALARM_TRIGGERED,
+    STATE_CLOSED,
+    STATE_HOME,
+    STATE_IDLE,
+    STATE_LOCKED,
+    STATE_LOCKING,
+    STATE_NOT_HOME,
+    STATE_OFF,
+    STATE_OK,
+    STATE_ON,
+    STATE_OPEN,
+    STATE_OPENING,
+    STATE_PAUSED,
+    STATE_PLAYING,
+    STATE_PROBLEM,
+    STATE_UNLOCKED,
+    STATE_UNLOCKING,
+    Platform,
+)
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.integration_platform import (
     async_process_integration_platforms,
 )
 
 from .const import DOMAIN, REG_KEY
 
+EXCLUDED_DOMAINS: set[Platform | str] = {
+    Platform.AIR_QUALITY,
+    Platform.SENSOR,
+    Platform.WEATHER,
+}
+
+ON_OFF_STATES: dict[Platform | str, tuple[set[str], str, str]] = {
```

Comment:
We should add comments here that these are considered immutable in respect that new platforms should be added, and the only maint here would be if existing platforms add new states.

Reply:
done